### PR TITLE
Improve Docs

### LINKED ISSUE #103: Improve documentation about `webpack-manifest-plugin-after-emit` hook.
It will be great, thanks!

### PR DESCRIPTION
**Pull request by a-x-**
- add use-cases for options
- closes #103 `webpack-manifest-plugin-after-emit` hook

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,13 +93,31 @@ Type: `function`
 
 Filter out files. [more details](#hooks-options)
 
+**Use-case**:  Omit [dll-chunks] ([issue](https://github.com/danethurber/webpack-manifest-plugin/issues/46)):
```
Comment (mastilver) [inline README.md]: It's not specific to dll chunks, also dynamic one (when you use `import(...).then(...)`)

File: README.md [hunk 2]
```diff
@@ -93,13 +93,31 @@ Type: `function`
...
 ### `options.map`
 
 Type: `function`
 
 Modify files details before the manifest is created. [more details](#hooks-options)
 
+**Use-case**: Add hash information to manifest file for [SRI] ([issue](https://github.com/danethurber/webpack-manifest-plugin/issues/35), [issue](https://github.com/danethurber/webpack-manifest-plugin/issues/55)):
+```js
+  map: ({chunk, file}) => ({ file, hash: chunk.source.integrity }),
+  generate: (seed, files) => files.reduce((manifest, {name, path}) => manifest), seed),
```
Comment (mastilver) [inline README.md]: I don't think that is working?

Comment (a-x-) [reply README.md]: I'm not sure too 😂
I just rewrited from `reduce` to `generate`

after:
```
map: ({chunk, file}) => {
  return {
    file,
    hash: chunk.source.integrity
  };
}

reduce: (manifest, {chunk, file}) => {
  return manifest;
}
```

default generate is:
```
`(seed, files) => files.reduce((manifest, {name, path}) => ({...manifest, [name]: path}), seed)`
```

**Comment by shellscape** [comment]
Thanks for opening this PR a while back, and I'm sorry that it didn't get attention sooner. We've updated the documentation as part of #222. If you'd like to add examples to the various methods, please open a new issue with your suggestions and we can talk about the concept of a `recipes` directory.

**Pull request closed without merging**